### PR TITLE
Use .slack-notifier-base on notify-slack job

### PR DESCRIPTION
**Pull request by KevinFairise2**
### What does this PR do?

Fix failures on deploy pipelines for `notify-slack` job

### Reviewer's Checklist

- [ ] If known, an appropriate milestone has been selected; otherwise the `Triage` milestone is set.
- [ ] Use the `major_change` label if your change either has a major impact on the code base, is impacting multiple teams or is changing important well-established internals of the Agent. This label will be use during QA to make sure each team pay extra attention to the changed behavior. For any customer facing change use a releasenote.
- [ ] A [release note](https://github.com/DataDog/datadog-agent/blob/main/docs/dev/contributing.md#reno) has been added or the `changelog/no-changelog` label has been applied.
- [ ] Changed code has automated tests for its functionality.
- [ ] Adequate QA/testing plan information is provided. Except if the `qa/skip-qa` label, with required either `qa/done` or `qa/no-code-change` labels, are applied.
- [ ] At least one `team/..` label has been applied, indicating the team(s) that should QA this change.
- [ ] If applicable, docs team has been notified or [an issue has been opened on the documentation repo](https://github.com/DataDog/documentation/issues/new).
- [ ] If applicable, the `need-change/operator` and `need-change/helm` labels have been applied.
- [ ] If applicable, the `k8s/<min-version>` label, indicating the lowest Kubernetes version compatible with this feature.
- [ ] If applicable, the [config template](https://github.com/DataDog/datadog-agent/blob/main/pkg/config/config_template.yaml) has been updated.

### CODE REVIEW

**Comment by KevinFairise2** [comment]
/trigger-ci --variable RUN_ALL_BUILDS=true --variable RUN_KITCHEN_TESTS=true --variable RUN_E2E_TESTS=auto --variable DEPLOY_AGENT=true

File: .gitlab/internal_kubernetes_deploy.yml [hunk 1]
```diff
@@ -51,7 +55,6 @@ notify-slack:
     - if: $DDR != "true"
       when: never
     - !reference [ .on_deploy_a7 ]
-  image: registry.ddbuild.io/slack-notifier:sdm
   tags: ["arch:amd64"]
```
Comment (chouquette) [inline .gitlab/internal_kubernetes_deploy.yml:58]: This is already part of the template and can be removed

File: .gitlab/internal_kubernetes_deploy.yml [hunk 2]
```diff
@@ -51,7 +55,6 @@ notify-slack:
...
   needs: ["internal_kubernetes_deploy_experimental"]
   script:
```
Comment (chouquette) [inline .gitlab/internal_kubernetes_deploy.yml:60]: Wouldn't keeping the `script` section prevent the template from doing what it needs to do?

Comment (KevinFairise2) [reply .gitlab/internal_kubernetes_deploy.yml:60]: Yes we do the same for `notify` job. The idea is to use the `.slack-notifier-base` to have the correct image and we can override the script and do what we want

File: .gitlab/internal_kubernetes_deploy.yml (hunk 1 above)
Comment (KevinFairise2) [reply .gitlab/internal_kubernetes_deploy.yml:58]: I see that we also redefine it in the the `notify` job: https://github.com/DataDog/datadog-agent/blob/main/.gitlab/notify.yml#L19-L41
So maybe this is clearer for the CI to keep the tag, if that's ok for you

File: .gitlab/internal_kubernetes_deploy.yml (hunk 2 above)
Comment (chouquette) [reply .gitlab/internal_kubernetes_deploy.yml:60]: ah indeed! Thanks for the clarification!

File: .gitlab/internal_kubernetes_deploy.yml (hunk 1 above)
Comment (chouquette) [reply .gitlab/internal_kubernetes_deploy.yml:58]: Definitely not a blocking comment :)

**Review by chouquette — APPROVED** [review]

**Pull request merged**